session/session.go

```diff
@@ -77,6 +101,9 @@ func (s *ZKSession) Subscribe(subscription chan<- ZKSessionEvent) {
 func (s *ZKSession) notifySubscribers(event ZKSessionEvent) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
+	if s.log != nil {
+		s.log.Printf("gozk-recipes/session: %s event occured", event.String())
+	}
```

Comment:
I'm sorry I didn't include this in the first review.. but it springs to mind that it's likely a better idea to log inline in `manage()`. It means you don't really need the `String()` and you can attach a little description to your event, e.g. in light of your previous comment it'd be nice to know that for insane reasons we hit `STATE_CLOSED` which should absolutely not happen and the `session` has stopped retrying to connect.